src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -1576,12 +1564,7 @@ public int GetHashCode(ReadOnlySpan<char> source, CompareOptions options)
                     return GetHashCodeOfStringCore(source, options);
                 }
 
-                if ((options & CompareOptions.IgnoreCase) == 0)
-                {
-                    return string.GetHashCode(source);
-                }
-
-                return string.GetHashCodeOrdinalIgnoreCase(source);
+                return InvariantGetHashCode(source, options);
```

Comment:
The method is called when `if (!GlobalizationMode.Invariant)`, so it is invariant way of hashing, based on the `CompareOptions` passed to the method. I do not see anything misleading in the name." There is also ignore-case considered." - it can be deduced looking at the arguments passed to the method